src/libponyc/expr/operator.c

```diff
@@ -549,6 +549,23 @@ bool expr_as(pass_opt_t* opt, ast_t** astp)
     return false;
   }
 
+  if(is_eqtype(expr_type, type, NULL, opt))
```

Comment:
Note that the definition of `is_eqtype` is equivalent to an `is_subtype` call in each direction:

```c
bool is_eqtype(ast_t* a, ast_t* b, errorframe_t* errorf, pass_opt_t* opt)
{
  return is_subtype(a, b, errorf, opt) && is_subtype(b, a, errorf, opt);
}
```

Therefore, this strategy of testing `is_eqtype`, then testing `is_subtype` is kind of wasteful - because we are doing three `is_subtype`checks every time, even in the "happy path". I've found that excess `is_subtype` calls can sometimes be very costly to compilation time and memory.

I'd suggest an approach of testing `is_subtype` first, then if that is true, continuing on to test `is_subtype` in the other direction to determine the appropriate error message. This approach would only have one `is_subtype` check in the "happy path", and two `is_subtype` checks in the error path.

Reply:
thank you!